### PR TITLE
Olwa pr

### PR DESCRIPTION
**Pull request by oliver-walker-hms**

### CODE REVIEW

**Comment by oliver-walker-hms** [comment]
My testing was all OMF centered until we got the new test environment setup. It turns out the other API is broken in the current release. This fully restores it to functionality.

**Review by TomKimsey — CHANGES_REQUESTED** [review]
Commit titles and messages dont describe why the changes are being made other than saying that they are "fixes". These commits need to explain the reason for the changes as some of the changes are hard to follow.

**Review by alexjhawk — CHANGES_REQUESTED** [review]
Commit titles/messages

File: src/com/hms/flexyosisoftconnector/OsisoftJsonPayload.java [hunk 1]
```diff
@@ -381,15 +381,34 @@ public void completePayloadAttempt() {
...
+            case OSIsoftConfig.OMF:
+              payload.append(PayloadBuilder.addContainerStartToOMFDataMessage(tagName, typeString));
+              payload.append(tagPayload.getPayload());
+              payload.append(PayloadBuilder.addContainerEndToOMFDataMessage());
+              break;
+            case OSIsoftConfig.PI_WEB_API:
+              payload.append(tagPayload.getPayload());
+              break;
+            default:
+              Logger.LOG_SERIOUS(comsErrMsg);
```
Comment (alexjhawk) [inline src/com/hms/flexyosisoftconnector/OsisoftJsonPayload.java:394]: Not in the scope of this PR, but worth noting that `comsErrMsg` looks like it is a constant, therefore it should be a `static final` and be capitalized appropriately using ALL_CAPS.

**Review by alexjhawk — APPROVED** [review]

File: src/com/hms/flexyosisoftconnector/OsisoftJsonPayload.java (hunk 1 above)
Comment (oliver-walker-hms) [reply src/com/hms/flexyosisoftconnector/OsisoftJsonPayload.java:394]: agreed, it should get changed to a static final in the next PR.

File: src/com/hms/flexyosisoftconnector/OSIsoftServer.java [hunk 2]
```diff
@@ -82,8 +81,9 @@ public static boolean RequestHTTPS(
       String FileFields,
       String FileName) {
     int res = NO_ERROR;
-    FileName = RESPONSE_DIRECTORY + FileName;
-
+    if (!FileName.equalsIgnoreCase("")) {
```
Comment (TomKimsey) [inline src/com/hms/flexyosisoftconnector/OSIsoftServer.java:84]: why not just check 
```
if (FileName.length() > 0)
```

**Review by TomKimsey — DISMISSED** [review]

File: src/com/hms/flexyosisoftconnector/OSIsoftServer.java (hunk 2 above)
Comment (TomKimsey) [reply src/com/hms/flexyosisoftconnector/OSIsoftServer.java:84]: Seems like an inefficient way to check that its not an empty string. if only we had 1.6 and could use String.isEmpty() :)

Comment (TomKimsey) [reply src/com/hms/flexyosisoftconnector/OSIsoftServer.java:84]: What happens if it is empty?

Comment (oliver-walker-hms) [reply src/com/hms/flexyosisoftconnector/OSIsoftServer.java:84]: I like FileName.length() > 0 as a check better, will change to that shortly

Comment (oliver-walker-hms) [reply src/com/hms/flexyosisoftconnector/OSIsoftServer.java]: Changed

Comment (oliver-walker-hms) [reply src/com/hms/flexyosisoftconnector/OSIsoftServer.java]: Without this if statement it would break being passed "/usr/responseDir/", but with it no responses are saved on empty file names. No response files getting saved is the desired behavior on empty filename strings

**Review by TomKimsey — APPROVED** [review]

**Review by alexjhawk — APPROVED** [review]
